### PR TITLE
Add an "md scan" command to kbfstool

### PR DESCRIPTION
**Pull request by akalin-keybase**
For now, it just checks for dangling blocks.

This is for diagnosing problems where blocks
are erroneously removed by e.g. conflict
resolution.

### CODE REVIEW

File: kbfstool/md_common.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/keybase/kbfs/fsrpc"
+	"github.com/keybase/kbfs/libkbfs"
+	"golang.org/x/net/context"
+)
+
+var mdGetRegexp = regexp.MustCompile("^(.+?)(?::(.*?))?(?:\\^(.*?))?$")
```
Comment (akalin-keybase) [inline kbfstool/md_common.go]: This is all just moved from `md_dump.go`

**Comment by akalin-keybase** [comment]
Note that this is based on top of #258, but can be reviewed independently.

File: kbfstool/md_main.go [hunk 2]
```diff
@@ -12,6 +12,7 @@ const mdUsageStr = `Usage:
 
 The possible subcommands are:
   dump		Dump metadata objects
+  check		Check metadata objects and their associated blocks for errors
```
Comment (strib) [inline kbfstool/md_main.go]: This looks like it's only downloading things, not really checking their true validity.  Especially for the MD object.  Should we call `BareRootMetadata.IsValidAndSigned()` in `mdGet`?

**Comment by strib** [comment]
Looks good to me.  Though did you know about `libkbfs.StateChecker.findAllBlocksInPath`?  Not a requirement, but it seems like this duplicates much of that functionality, so it might be worth seeing if we can refactor a bit to have them share code.  But the state checker is also building up block sizes, so it's not quite a perfect fit.

File: kbfstool/md_main.go (hunk 2 above)
Comment (akalin-keybase) [inline kbfstool/md_main.go]: I was going to comment that `IsValidAndSigned` is already called by `MDOps`, but that wasn't quite true! Rebased this on top of https://github.com/keybase/kbfs/pull/263 , which makes that true, and added comments.

**Comment by akalin-keybase** [comment]
I left a TODO to decomp common code with libkbfs.StateChecker.findAllBlocksInPath.

**Pull request merged**